ui/app/src/components/AwaitingApprovalDashlet/AwaitingApprovalDashlet.tsx

```diff
@@ -193,11 +201,50 @@ export default function AwaitingApprovalDashlet(props: AwaitingApprovalDashletPr
   };
 
   const onToggleCheckedAll = () => {
-    onItemChecked(Object.keys(state.itemsLookup), !isAllChecked);
+    if (isAllChecked) {
+      setSelectedLookup({});
+    } else {
+      setSelectedLookup({ ...selectedLookup, ...createPresenceTable(Object.keys(state.itemsLookup), true) });
+    }
   };
 
   const handleItemChecked = (path: string) => {
-    onItemChecked([path]);
+    setSelectedLookup({ ...selectedLookup, [path]: !selectedLookup[path] });
+  };
+
+  const onItemMenuClick = (event: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement>, item: DetailedItem) => {
+    const path = item.path;
+    dispatch(
+      batchActions([
+        completeDetailedItem({ path }),
+        showItemMegaMenu({
+          path,
+          anchorReference: 'anchorPosition',
+          anchorPosition: { top: event.clientY, left: event.clientX },
+          numOfLoaderItems: getNumOfMenuOptionsForItem({
+            path: item.path,
+            systemType: getSystemTypeFromPath(item.path)
+          } as DetailedItem)
+        })
+      ])
+    );
+  };
+
+  const onActionBarOptionClicked = (option: string) => {
+    if (option === 'clear') {
+      setSelectedLookup({});
+    } else {
+      itemActionDispatcher({
+        site,
+        item: Object.keys(selectedLookup)
+          .filter((path) => selectedLookup[path])
+          .map((path) => state.itemsLookup[path]),
```

Comment:
Don't understand the filtering of selected lookup against it's own? Technically, that filter will always return the full array, no?
```suggestion
        item: Object.keys(selectedLookup)
          .map((path) => state.itemsLookup[path]),
```